src/impl_pipeline.cpp

```diff
@@ -378,48 +364,75 @@ namespace daxa
+                std::vector<u32> spirv = spirv_result.value();
+
+                VkShaderModuleCreateInfo vk_shader_module_create_info{
+                    .sType = VK_STRUCTURE_TYPE_SHADER_MODULE_CREATE_INFO,
+                    .pNext = nullptr,
+                    .codeSize = static_cast<u32>(spirv.size() * sizeof(u32)),
+                    .pCode = spirv.data(),
+                };
+
+                vkCreateShaderModule(impl.impl_device.as<ImplDevice>()->vk_device, &vk_shader_module_create_info, nullptr, &vk_shader_module);
+
+                neeeded_shader_stages++;
+            }
+
+            return true;
         };
-        vkCreateShaderModule(impl.impl_device.as<ImplDevice>()->vk_device, &shader_module_pixel, nullptr, &p_vk_shader_module);
+
+        VkShaderModule v_vk_shader_module = {};
+
+        {
+            auto result = create_shader_module(modified_info.vertex_shader_info, VkShaderStageFlagBits::VK_SHADER_STAGE_VERTEX_BIT, v_vk_shader_module);
+            if(result.is_err()) {
+                return ResultErr{.message = result.message()};
+            } 
+        }
+
+        VkShaderModule p_vk_shader_module = {};
+
+        {
+            auto result = create_shader_module(modified_info.fragment_shader_info, VkShaderStageFlagBits::VK_SHADER_STAGE_FRAGMENT_BIT, p_vk_shader_module);
+            if(result.is_err()) {
+                return ResultErr{.message = result.message()};
+            } 
+        }
 
         impl_pipeline->vk_pipeline_layout = impl.impl_device.as<ImplDevice>()->gpu_table.pipeline_layouts[(modified_info.push_constant_size + 3) / 4];
 
-        VkPipelineShaderStageCreateInfo vk_pipeline_shader_stage_create_infos[2] = {
-            VkPipelineShaderStageCreateInfo{
-                .sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO,
-                .pNext = nullptr,
-                .flags = {},
-                .stage = VkShaderStageFlagBits::VK_SHADER_STAGE_VERTEX_BIT,
-                .module = v_vk_shader_module,
-                .pName = modified_info.vertex_shader_info.compile_options.entry_point.value().c_str(),
-                .pSpecializationInfo = nullptr,
-            },
-            VkPipelineShaderStageCreateInfo{
-                .sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO,
-                .pNext = nullptr,
-                .flags = {},
-                .stage = VkShaderStageFlagBits::VK_SHADER_STAGE_FRAGMENT_BIT,
-                .module = p_vk_shader_module,
-                .pName = modified_info.fragment_shader_info.compile_options.entry_point.value().c_str(),
-                .pSpecializationInfo = nullptr,
-            },
+        std::vector<VkPipelineShaderStageCreateInfo> vk_pipeline_shader_stage_create_infos;
+        vk_pipeline_shader_stage_create_infos.reserve(neeeded_shader_stages);
+
+        auto create_shader_stage = [&](ShaderInfo const & shader_info, VkShaderStageFlagBits shader_stage, VkShaderModule& vk_shader_module) {
```

Comment:
have the vector created earlier and push_back those at the same time as the shader modules